nio/service/base.py

```diff
@@ -198,13 +216,21 @@ def configure(self, context):
                 configure_threads.append(
                     spawn(block.do_configure, block_context))
             else:
-                block.do_configure(block_context)
+                try:
+                    block.do_configure(block_context)
+                except Exception as e:
+                    raise BlockException(e, label=block.label())
             # register it
             self._blocks[block.id()] = block
         # if configuration was async, ensure they are all done
         if configure_threads:
             for thread in configure_threads:
-                thread.join()
+                try:
+                    thread.join()
+                except Exception as e:
+                    label = thread._args[0].properties["name"] or \
```

Comment:
I think we should change this to be similar to the syntax for the other async execution, it seems like it should be possible. So make the thread list contain the block reference as well rather than diving in to the thread arguments

Reply:
:+1: